### ISSUE TITLE
[eslint-plugin-nx] Support ESLint 8.x

### ISSUE THREAD

**Issue by MichaelDeBoey** [nrwl/nx#6866]
ESLint v8.0.0 is [released](https://eslint.org/blog/2021/10/eslint-v8.0.0-released) 🎉

It would be awesome to have official ESLint 8 support. 👊
I'm happy to help where I can of course 🙂

**Comment by JamesHenry** [comment]
Hi @MichaelDeBoey would you mind clarifying what you are requesting here? That ESLint 8 is supported while it is still in prerelease?

If so I don't think that is feasible for Nx - ESLint 8 being in beta means breaking changes are still occurring, and Brad's PR for v5 of typescript-eslint is having to keep up with that.

Adding a third level of separation with Nx on top and I don't think it is feasible to support two different implementations to support across that (still evolving) breaking changes.

If I have misunderstood, please let me know! Naturally once ESLint v8 is the stable version we will certainly be adding full support for it to Nx

**Comment by JamesHenry** [comment]
Additionally, are there specific roadblocks you have run into when applying changes to your Nx workspace to support ESLint v8 that having support in Nx itself unblocks?

**Comment by puku0x** [comment]
The error looks similar to https://github.com/typescript-eslint/typescript-eslint/issues/3735

**Comment by mheob** [comment]
What is the current status here? I get the same error as @alfaproject (https://github.com/nrwl/nx/issues/6866#issuecomment-946816766).

**Comment by ramsesDeco** [comment]
> The Nx team will deploy it soon, you can see this comment for more details

https://github.com/angular-eslint/angular-eslint/issues/843#issuecomment-977008811

**Comment by NachoVazquez** [comment]
> > The Nx team will deploy it soon, you can see this comment for more details
> 
> 
> 
> https://github.com/angular-eslint/angular-eslint/issues/843#issuecomment-977008811

Thank you! I misunderstood that it was already present in v12.2.2

**Comment by huantaoliu** [comment]
After updating @nrwl/eslint-plugin-nx from 14.0.3 -> 14.0.5, the problem seems happen again
```
Failed to load plugin '@nrwl/nx' declared in 'libs/nx-lib/.eslintrc.json » ../../.eslintrc.json#overrides[0]': Class extends value undefined is not a constructor or null
Referenced from: /builds/the-mill/lambda-gaming/.eslintrc.json
```

**Comment by HashemKhalifa** [comment]
still facing the same issue after migrating to `14.6.4`

**Comment by meeroslav** [comment]
@taletski Nx does not support PnP officially. We have only tested Nx with Yarn Berry using node_modules linker. The support for PnP has been on our list, but currently has low priority.

Can you please create a minimal repo that shows this error, as potentially it can be fixed quick. And please create a new issue, since it doesn't match the original request.
